### PR TITLE
Adds support for new iOS 13 traits

### PR DESCRIPTION
**Pull request by ay8s**
Adds userInterfaceLevel, accessibilityContrast and legibilityWeight available in iOS 13. 

While looking into this I spotted that NSStringFromASPrimitiveTraitCollection (ASTraitCollection.mm) is inserting a nil value for preferredContentSizeCategory under iOS 13 when rotating a device. Similar to what we were seeing here... https://github.com/TextureGroup/Texture/issues/754

We could create a AS_NSString method to handle it or check if it doesn't exist and just pass add UIContentSizeCategoryUnspecified.

### CODE REVIEW

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/2e561e573bb28ddeea0041d696cb9bd9/raw/16a2df7f2b76105c74446f9958f86cb098e243fb/log.txt)

File: Source/Details/ASTraitCollection.h [hunk 1]
```diff
@@ -48,6 +48,10 @@ typedef struct {
   unowned UIContentSizeCategory preferredContentSizeCategory API_AVAILABLE(ios(10.0));
 
   CGSize containerSize;
+
```
Comment (rahul-malik) [inline Source/Details/ASTraitCollection.h]: Do these changes compile in Xcode 10? We should probably make this work in both the Xcode 11 beta and Xcode 10 GM until we have a GM version of Xcode 11

**Comment by nguyenhuy** [comment]
Note to reviewers: Blocked by Xcode 11 GM and CI upgrade.

**Comment by noamalffasy** [comment]
@nguyenhuy Xcode 11 GM is out now

**Comment by rahul-malik** [comment]
@ay8s - Hey just checking in on this since it'll be helpful for folks building for iOS 13. Can you rebase / resolve conflicts?

**Comment by ay8s** [comment]
@rahul-malik Away on a retreat this week so can take a look next week. Feel free to pick up if you need it sooner.

**Comment by rahul-malik** [comment]
@ay8s - Ping :)

**Comment by nguyenhuy** [comment]
FYI, I think this PR causes the podspec linting CI job to [fail](https://pipelines.actions.githubusercontent.com/1q9zsgiSRXplxbeCjr3wjoUubhp2khLVKBKsvj0siPSc80rGF9/_apis/pipelines/1/runs/856/signedlogcontent/3?urlExpires=2020-01-02T22%3A29%3A39.7401679Z&urlSigningMethod=HMACV1&urlSignature=4w1CqvzscgOUGOF%2Be%2BZTx4cNujlwd47BRkdj8xZI%2Buw%3D):

```
- ERROR | [Texture/Core, Texture/PINRemoteImage, Texture/IGListKit, and more...] xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.

- ERROR | [Texture/Core, Texture/PINRemoteImage, Texture/IGListKit, and more...] xcodebuild:  /Users/runner/runners/2.163.1/work/Texture/Texture/Source/Details/ASTraitCollection.h:51:3: error: 'UIUserInterfaceLevel' is unavailable: not available on tvOS

- ERROR | [Texture/Core, Texture/PINRemoteImage, Texture/IGListKit, and more...] xcodebuild:  /Users/runner/runners/2.163.1/work/Texture/Texture/Source/Details/ASTraitCollection.h:155:22: error: 'UIUserInterfaceLevel' is unavailable: not available on tvOS

- ERROR | [tvOS] [Texture/TextNode2] xcodebuild:  /Users/runner/runners/2.163.1/work/Texture/Texture/Source/Details/ASTraitCollection.h:155:22: error: 
- NOTE  | [tvOS] [Texture/TextNode2] xcodebuild:  note: 'setInsetsLayoutMarginsFromSafeArea:' has been marked as being introduced in tvOS 11.0 here, but the deployment target is tvOS 9.0.0
```

https://github.com/TextureGroup/Texture/runs/371568518

https://pipelines.actions.githubusercontent.com/1q9zsgiSRXplxbeCjr3wjoUubhp2khLVKBKsvj0siPSc80rGF9/_apis/pipelines/1/runs/856/signedlogcontent/3?urlExpires=2020-01-02T22%3A32%3A11.8909147Z&urlSigningMethod=HMACV1&urlSignature=2rtQHGXFK2zhDzQdWmM1wri06MbCHCPuv5u5EGM1XZE%3D

**Comment by jparise** [comment]
> Perhaps we should run podspec lint on every pull request to catch issues earlier? Right now it's run on master only because the task is really slow.

Good idea. I think it would be worth it.

**Pull request merged**